plugin/src/main/java/org/wildfly/plugins/bootablejar/maven/goals/DevWatchBootableJarMojo.java

```diff
@@ -293,6 +338,131 @@ ModelNode waitStatus(ModelControllerClient client, String status, ModelNode op)
         }
     }
 
+    private class RemoteDeploymentController implements DeploymentController {
+        private final String name;
+
+        public RemoteDeploymentController(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public void deploy(Path dir) throws Exception {
```

Comment:
I think that we should share the logic with the localDeploymentController, introduce an AbstractDeploymentController?